data/code/lib.c

```diff
@@ -216,3 +217,186 @@ int itemshufflemgr_calcslot_ex(const unsigned int* const itemshufflemgr,
 }
 
 #endif
+
+#if SECTIONED_COURSES
+
+static char passed;
+static unsigned int sectionCount = 0;
+
+// override the lap count in a section course to be the number of section points
+void sectionLapOverride()
+{
+    register unsigned short reg3 asm ("r3");
+    register unsigned short reg9 asm ("r9");
+    asm("or %r0,%r3,%r3 \t\n");
+
+    reg9 = (unsigned short)sectionCount;
+
+    if (reg9 != 0)
+    {
+        if (reg9 > 9)
+        {
+            asm("li %r10, 0x9       \t\n"
+                "sth %r10,0x2e(%r31) \t\n");
+        }
+        else
+        {
+            asm("or %r10, %r9, %r9  \t\n"
+                "sth %r10,0x2e(%r31) \t\n");
+        }
+    }
+}
+
+void resetSectionCount()
+{
+    asm("or %r31, %r3, %r3  \t\n");             // do this first
+    sectionCount = 0;
+}
+
+// count the section points to jury-rig the "max laps" count to always be the section number
+void countSectionPoints()
+{
+    register unsigned int *reg0 asm ("r0");
+    register unsigned int reg3 asm ("r3");
```

Comment:
I understand that `r0` and `r3` are declared here to enforce preservation, as otherwise the compiler may use them. If that's the case, can we use the default implicit type (`int`) for these cases?